### PR TITLE
fix(gatsby): fix Uncaught TypeError in navigation.js

### LINKED ISSUE #21263: navigation.js:173 Uncaught TypeError
Still getting this error from links after updating to Gatsby 2.19.12. I've checked my dependencies and re-installed node modules.

I can recreate this issue by:

1. updating [this](https://github.com/thomaswang/gatsby-personal-starter-blog) starter
2. making a new root element implementing the router with '/' corresponding to 'index.js', '/foo' corresponding to the blog or some trivial page
3. putting a new link to the blog or some trivial page onto index.js and clicking it

```
navigation.js:173 Uncaught TypeError: Cannot read property 'getElementsByTagName' of null
    at navigation.js:173
```

```
>  170 | if (document.title) {
>  171 |      pageName = document.title
>  172 | }
>  173 | const pageHeadings = document
>  174 |     .getElementById(`gatsby-focus-wrapper`)
>  175 |     .getElementsByTagName(`h1`)
>  176 | if (pageHeadings && pageHeadings.length) {
```

See [this previously closed issue](https://github.com/gatsbyjs/gatsby/issues/21139), seems other people are still running into this aswell

### PR DESCRIPTION
**Pull request by shicholas**
## Description

https://github.com/gatsbyjs/gatsby/issues/21263 is an issue many people are arriving at for a variety of reasons.  Since the code contains this logic on the following line:

```
      if (pageHeadings && pageHeadings.length) {
```

There should be no issue in seeing if `document.getElementById(`gatsby-focus-wrapper`) exists first before attempting to look up `getElementsByTagName`.

### Documentation

This was a new change document in #21263 and #21139 

## Related Issues

Fixes #21263

### CODE REVIEW

File: packages/gatsby/cache-dir/navigation.js [hunk 1]
```diff
@@ -172,6 +172,8 @@ class RouteAnnouncer extends React.Component {
         pageName = document.title
       }
       const pageHeadings = document
+        .getElementById(`gatsby-focus-wrapper`) &&
+        document
         .getElementById(`gatsby-focus-wrapper`)
         .getElementsByTagName(`h1`)
```
Comment (vladar) [inline packages/gatsby/cache-dir/navigation.js:178]: Maybe we can simply do this:

```suggestion
      const pageHeadings = document
        .querySelectorAll(`#gatsby-focus-wrapper h1`)
```

@madalynrose What's your opinion on this?

Comment (shicholas) [reply packages/gatsby/cache-dir/navigation.js:178]: fwiw, looks better to me than what I had.

Comment (madalynrose) [reply packages/gatsby/cache-dir/navigation.js]: yeah that looks good to me. nice solution @vladar

**Comment by saadaouad** [comment]
Could you please release this changes guys if it looks good for you?

**Comment by vladar** [comment]
Published in `gatsby@2.20.29`

**Pull request merged**